src/v/cloud_storage/partition_manifest.cc

```diff
@@ -175,6 +175,33 @@ segment_name generate_local_segment_name(model::offset o, model::term_id t) {
     return segment_name(ssx::sformat("{}-{}-v1.log", o(), t()));
 }
 
+partition_manifest::lw_segment_meta
+partition_manifest::lw_segment_meta::convert(const segment_meta& m) {
+    return lw_segment_meta{
+      .ntp_revision = m.ntp_revision,
+      .base_offset = m.base_offset,
+      .committed_offset = m.committed_offset,
+      .archiver_term = m.archiver_term,
+      .segment_term = m.segment_term,
+      .size_bytes = m.sname_format == segment_name_format::v2 ? m.size_bytes
+                                                              : 0,
+    };
+}
+
+segment_meta partition_manifest::lw_segment_meta::convert(
```

Comment:
I get why we convert from a segment meta to a low_segment_meta (when moving to _replaced), but why do we need the opposite direction?

Reply:
removed that

Reply:
actually, it's used in `partition_manifest::replaced_segments` which returns the list of `segment_meta` structs